src/hu/openig/scripting/missions/Mission21.java

```diff
@@ -266,10 +267,16 @@ public void onAutobattleFinish(BattleInfo battle) {
 	 */
 	boolean concludeBattle(BattleInfo battle) {
 		boolean result = false;
+		boolean prototypeSurvived = true;
 		clearMission("Mission-21-Timeout");
-		Fleet carrier = findTaggedFleet(ALLY, player);
 
-		if (carrier != null) {
+		for (SpacewarStructure s: battle.spaceLosses) {

```

Comment:
I see now that `tagFleet` is already tagging everything in the fleet with the value of `ALLY`, which means even the fighters would have that tag. I think I'll have to find another solution.